java/ql/lib/semmle/code/java/frameworks/MyBatis.qll

```diff
@@ -105,94 +105,88 @@ class TypeParam extends Interface {
   TypeParam() { this.hasQualifiedName("org.apache.ibatis.annotations", "Param") }
 }
 
-module ProviderInjection {
-  private class MyBatisAbstractSQL extends RefType {
-    MyBatisAbstractSQL() { this.hasQualifiedName("org.apache.ibatis.jdbc", "AbstractSQL") }
-  }
-
-  private class MyBatisProvider extends RefType {
-    MyBatisProvider() {
-      this.hasQualifiedName("org.apache.ibatis.annotations",
-        ["Select", "Delete", "Insert", "Update"] + "Provider")
-    }
-  }
+private class MyBatisAbstractSQL extends RefType {
+  MyBatisAbstractSQL() { this.hasQualifiedName("org.apache.ibatis.jdbc", "AbstractSQL") }
+}
 
-  private class MyBatisAbstractSQLMethodNames extends string {
-    MyBatisAbstractSQLMethodNames() {
-      this in [
-          "SELECT", "OFFSET_ROWS", "FETCH_FIRST_ROWS_ONLY", "OFFSET", "LIMIT", "ORDER_BY", "HAVING",
-          "GROUP_BY", "WHERE", "OUTER_JOIN", "RIGHT_OUTER_JOIN", "LEFT_OUTER_JOIN", "INNER_JOIN",
-          "JOIN", "FROM", "DELETE_FROM", "SELECT_DISTINCT", "SELECT", "INTO_VALUES", "INTO_COLUMNS",
-          "VALUES", "INSERT_INTO", "SET", "UPDATE"
-        ]
-    }
+private class MyBatisProvider extends RefType {
+  MyBatisProvider() {
+    this.hasQualifiedName("org.apache.ibatis.annotations",
+      ["Select", "Delete", "Insert", "Update"] + "Provider")
   }
+}
 
-  class MyBatisInjectionSink extends DataFlow::Node {
-    MyBatisInjectionSink() {
-      exists(Annotation a, Method m, TypeLiteral type, Class c |
-        a.getType() instanceof MyBatisProvider and
-        type = a.getValue(["type", "value"]) and
-        c.hasMethod(m, type.getTypeName().getType()) and
-        m.hasName(a.getValue("method").(StringLiteral).getValue()) and
-        this.asExpr() = m.getBody().getAStmt().(ReturnStmt).getResult()
-      )
-    }
+private class MyBatisAbstractSQLMethodNames extends string {
+  MyBatisAbstractSQLMethodNames() {
+    this in [
+        "SELECT", "OFFSET_ROWS", "FETCH_FIRST_ROWS_ONLY", "OFFSET", "LIMIT", "ORDER_BY", "HAVING",
+        "GROUP_BY", "WHERE", "OUTER_JOIN", "RIGHT_OUTER_JOIN", "LEFT_OUTER_JOIN", "INNER_JOIN",
+        "JOIN", "FROM", "DELETE_FROM", "SELECT_DISTINCT", "SELECT", "INTO_VALUES", "INTO_COLUMNS",
+        "VALUES", "INSERT_INTO", "SET", "UPDATE"
+      ]
   }
+}
 
-  class MyBatisAdditionalTaintStep extends TaintTracking::AdditionalTaintStep {
-    abstract override predicate step(DataFlow::Node node1, DataFlow::Node node2);
+class MyBatisInjectionSink extends DataFlow::Node {
+  MyBatisInjectionSink() {
+    exists(Annotation a, Method m, TypeLiteral type, Class c |
+      a.getType() instanceof MyBatisProvider and
+      type = a.getValue(["type", "value"]) and
+      c.hasMethod(m, type.getTypeName().getType()) and
+      m.hasName(a.getValue("method").(StringLiteral).getValue()) and
+      this.asExpr() = m.getBody().getAStmt().(ReturnStmt).getResult()
+    )
```

Comment:
I think this can be simplified:

```suggestion
    exists(Annotation a, Method m |
      a.getType() instanceof MyBatisProvider and
      m.getDeclaringType() = a.getValue(["type", "value"]).(TypeLiteral).getTypeName().getType() and
      m.hasName(a.getValue("method").(StringLiteral).getValue()) and
      this.asExpr() = m.getBody().getAStmt().(ReturnStmt).getResult()
    )
```